### ISSUE TITLE
Autocompletion rules for `pkg::` should show functions before data objects

### ISSUE THREAD

**Issue by DavisVaughan** [rstudio/rstudio#12678]
In recent versions of the IDE, `pkg::<TAB>` now shows data objects before function names. I think this is fairly unintuitive because my 99% case is reaching for a function name.

It also breaks one of my autocompletion muscle memory workflows for calling `pak::pak()` 😅 

<img width="536" alt="Screen Shot 2023-02-07 at 1 01 59 PM" src="https://user-images.githubusercontent.com/19150088/217330176-dfb6b7dd-414c-4ab7-8810-b1bf47b7115a.png">

It used to autocomplete `pak::pak()` first so I could just hit tab after typing out `pak::pak` (or something like that).

RStudio 2023.03.0-daily+293 "Cherry Blossom" Daily (df7efaa7c605d91399201ad68eb607d5c16fda54, 2023-02-06) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.03.0-daily+293 Chrome/108.0.5359.179 Electron/22.0.3 Safari/537.36

**Comment by ronblum** [comment]
Does this mean that it would preferable to keep the order completely alphabetical, regardless of type, the way it was previously?

**Comment by DavisVaughan** [comment]
That would be fine with me.

Or data objects last.

I think I've just convinced myself that unconditionally putting data objects _first_ is the main thing that's confusing here.

**Comment by ronblum** [comment]
As a non-R-programmer, I won't inject my opinion. 😄  I'll leave that part to you, @kevinushey, and @romainfrancois. I'll just do the testing.

**Comment by romainfrancois** [comment]
This is just when completing after a `<pkg>::<TAB>` right ? i.e. should data frame still appear first in normal completions ?

**Comment by DavisVaughan** [comment]
Oh like this? Yes I think in this case it is reasonable to prefer the user's local variables

<img width="481" alt="Screen Shot 2023-02-08 at 11 01 32 AM" src="https://user-images.githubusercontent.com/19150088/217583422-f8f87cfd-098f-4b92-8931-24c5014539d7.png">

**Comment by DavisVaughan** [comment]
I'd be fine with:
- Local variables first
- _Anything_ from a package shows up in alphabetical order

**Comment by jonvanausdeln** [comment]
Tested on 2023.03.0-daily+323 
Windows 11

The `pak::p<tab>` case works as expected, showing the package elements in alphabetical order.

However, I believe this introduced a problem.  It displays a waring message for some actions in the console (see below reprex and screenshot)

1. Start IDE
2. in console, check version `RStudio.Version()`

This generates a warning message
<img width="464" alt="image" src="https://user-images.githubusercontent.com/6433789/219443865-f651365f-1857-461c-af32-005cff75854a.png">

Many things generate this message, but this was an easy reprex.  

e..g.
<img width="464" alt="image" src="https://user-images.githubusercontent.com/6433789/219444450-5c2620f8-ae51-4821-b9e2-225b4ce15e61.png">

I also verified that the build before this merge did not have this issue.
